internal/testutils/golden/golden.go

```diff
@@ -33,18 +33,18 @@ func init() {
 	}
 }
 
-type goldenOptions struct {
+type config struct {
```

Comment:
The type is not just used for the default config but also the config which is modified by the `Option` functions, so I prefer `config` over `defaults` (`default` is a reserved keyword so that can't be used anyway).